experimental/LieAlgebras/src/LieSubalgebra.jl

```diff
@@ -0,0 +1,296 @@
+  ) where {C<:RingElement,LieT<:LieAlgebraElem{C}}
+    return LieSubalgebra{C,LieT}(L, Vector{LieT}(map(L, gens)); kwargs...)
+  end
+end
+
+###############################################################################
+#
+#   Basic manipulation
+#
+###############################################################################
+
+base_lie_algebra(S::LieSubalgebra{C,LieT}) where {C<:RingElement,LieT<:LieAlgebraElem{C}} =
+  S.base_lie_algebra::parent_type(LieT)
+
+function gens(S::LieSubalgebra)
+  return S.gens
+end
+
+function gen(S::LieSubalgebra, i::Int)
+  return S.gens[i]
+end
+
+function ngens(S::LieSubalgebra)
+  return length(gens(S))
+end
+
+function basis_matrix(
+  S::LieSubalgebra{C,LieT}
+) where {C<:RingElement,LieT<:LieAlgebraElem{C}}
+  return S.basis_matrix::dense_matrix_type(C)
+end
+
+@doc raw"""
+    basis(S::LieSubalgebra{C}) -> Vector{LieAlgebraElem{C}}
+
+Return a basis of the Lie subalgebra `S`.
+"""
+function basis(S::LieSubalgebra)
+  return S.basis_elems
+end
+
+@doc raw"""
+    basis(S::LieSubalgebra{C}, i::Int) -> LieAlgebraElem{C}
+
+Return the `i`-th basis element of the Lie subalgebra `S`.
+"""
+function basis(S::LieSubalgebra, i::Int)
+  return S.basis_elems[i]
+end
+
+@doc raw"""
+  dim(S::LieSubalgebra) -> Int
+
+Return the dimension of the Lie subalgebra `S`.
+"""
+dim(S::LieSubalgebra) = length(basis(S))
+
+###############################################################################
+#
+#   String I/O
+#
+###############################################################################
+
+function Base.show(io::IO, ::MIME"text/plain", S::LieSubalgebra)
+  io = pretty(io)
+  println(io, LowercaseOff(), "Lie subalgebra")
+  println(io, Indent(), "of dimension $(dim(S))", Dedent())
+  if dim(S) != ngens(S)
+    println(io, Indent(), "with $(ItemQuantity(ngens(S), "generator"))", Dedent())
+  end
+  print(io, "of ")
+  print(io, Lowercase(), base_lie_algebra(S))
+end
+
+function Base.show(io::IO, S::LieSubalgebra)
+  io = pretty(io)
+  if get(io, :supercompact, false)
+    print(io, LowercaseOff(), "Lie subalgebra")
+  else
+    print(io, LowercaseOff(), "Lie subalgebra of ", Lowercase())
+    print(IOContext(io, :supercompact => true), base_lie_algebra(S))
+  end
+end
+
+###############################################################################
+#
+#   Comparisons
+#
+###############################################################################
+
+function Base.issubset(
+  S1::LieSubalgebra{C,LieT}, S2::LieSubalgebra{C,LieT}
+) where {C<:RingElement,LieT<:LieAlgebraElem{C}}
+  @req base_lie_algebra(S1) === base_lie_algebra(S2) "Incompatible Lie algebras."
+  return all(in(S2), gens(S1))
+end
+
+function Base.:(==)(
+  S1::LieSubalgebra{C,LieT}, S2::LieSubalgebra{C,LieT}
+) where {C<:RingElement,LieT<:LieAlgebraElem{C}}
+  base_lie_algebra(S1) === base_lie_algebra(S2) || return false
+  gens(S1) == gens(S2) && return true
+  return basis_matrix(S1) == basis_matrix(S2)
```

Comment:
Really? If the subalgebras have been constructed with `is_basis == true` then just the given bases are stored.
Are they necessarily equal if they span the same subalgebra?
(If yes then perhaps the term `is_basis` should perhaps be changed to `is_canonical_basis`.)

Reply:
This implementation of `==` is just testing equality as julia objects, not of mathematical objects. A change is added in 2843a5b686aa814a0522fac15b5f0149614e5449.